src/java/com/puppetlabs/jruby_utils/pool/JRubyPool.java

```diff
@@ -491,8 +491,7 @@ private void signalIfLockCanProceed() {
         // is active at a time - a caller of lock() that has just acquired
         // the pool lock but is waiting for all registered elements to be
         // returned to the queue.
-        if (registeredElements.size() == liveQueue.size() ||
-                pill != null) {
+        if (this.maxSize == liveQueue.size() || pill != null) {
```

Comment:
In any event, I agree that the comment in the code should be updated to talk about the max size/capacity of the pool rather than registered elements.